send-emails-from-a-service/tests/test.bal

```diff
@@ -0,0 +1,251 @@
+            "dob": "1940-03-19",
+            "ssn": "234-23-525",
+            "address": "California",
+            "phone": "8770586755",
+            "email": "johndoe@gmail.com",
+            "cardNo": "7844481124110331"
+        },
+        "doctor": "thomas collins",
+        "hospital_id": "grandoaks",
+        "hospital": "grand oak community hospital",
+        "appointment_date": "2023-10-02"
+    });
+
+    if resp !is http:ClientRequestError {
+        test:assertFail("expected an http:ClientRequestError, found " + (typeof resp).toString());
+    }
+
+    test:assertEquals(resp.message(), "Not Found");
+    var detail = resp.detail();
+    test:assertEquals(detail.statusCode, http:STATUS_NOT_FOUND);
+    test:assertEquals(detail.body, "unknown hospital, doctor, or category");
+}
+
+@test:Config
+function testEmptyEmailAddress() returns error? {
+    ReservationResponse|http:ClientError resp = cl->/surgery/reserve.post({
+        patient: {
+            name: "John Doe",
+            dob: "1940-03-19",
+            ssn: "234-23-525",
+            address: "California",
+            phone: "8770586755",
+            email: "",
+            cardNo: "7844481124110331"
+        },
+        doctor: "thomas collins",
+        hospital: "grand oak community hospital",
+        hospital_id: "grandoaks",
+        appointment_date: "2025-04-02"
+    });
+
+    if resp !is http:RemoteServerError {
+        test:assertFail("expected an http:ClientRequestError, found " + (typeof resp).toString());
+    }
+
+    test:assertEquals(resp.message(), "Internal Server Error");
+}
+
+isolated Appointment? & readonly appointment = ();
+
+public client class MockHttpClient {
+    private final string url;
+
+    isolated function init(string url) {
+        self.url = url;
+    }
+
+    isolated resource function post [http:PathParamType... path](http:RequestMessage message, 
+            map<string|string[]>? headers = (), 
+            string? mediaType = (), 
+            http:TargetType targetType = <>, *http:QueryParams params) 
+            returns http:Response|anydata|http:ClientError {
+        if self.url == hospitalServicesBackend {
+            return handleAppointment(path, message);
+        }
+
+        if self.url == paymentBackend {
+            return handlePayment(message);
+        }
+
+        return <http:ClientRequestError>error("unexpected request",
+                                            body = "unexpected request",
+                                            headers = {},
+                                            statusCode = http:STATUS_BAD_REQUEST);
+    }
+
+}
+
+isolated string? emailContent = "";
+
+public isolated client class MockEmailClient {
+    remote isolated function sendMessage(email:Message message) returns email:Error? {
+        lock {
+            emailContent = message.body;
+        }
+        if message.to == "" {
+            return error("invalid email address");
+        }
```

Comment:
Do we really need this? Not the only invalid email address, right?

Reply:
Thought it would be better check if the email address is empty or not